suave/core/utils.go

```diff
@@ -2,8 +2,33 @@ package suave
 
 import (
 	"encoding/json"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/google/uuid"
 )
 
+var bidUuidSpace = uuid.UUID{0x42}
+var emptyId [16]byte
+
+func calculateBidId(bid types.Bid) (types.BidId, error) {
+	copy(bid.Id[:], emptyId[:])
+
+	body, err := json.Marshal(bid)
```

Comment:
Not a blocker but I would consider doing `keccak(rlp(bid))` instead to have the same standard as the other hashed objects in the client.

Reply:
I'd defer to a later point, probably once we'll consider having another implementation - right now the id is localized to this function, and I'm not sure if bid will rlp-encode straight away